mep3_simulation/mep3_simulation/resistance_meter_driver.py

```diff
@@ -0,0 +1,179 @@
+        self.measuring_side = properties['measuringSide']
+
+        assert self.measuring_side in ['left', 'right']
+
+        self.__supervisor = webots_node.robot
+        self.__robot = self.__supervisor.getSelf()
+
+        self.__touch_sensor_front = self.__supervisor.getDevice(
+            f'hand_{self.measuring_side}_Dz_touch_sensor_front'
+        )
+        self.__touch_sensor_back = self.__supervisor.getDevice(
+            f'hand_{self.measuring_side}_Dz_touch_sensor_back'
+        )
+
+        self.__touch_sensor_front.enable(1)
+        self.__touch_sensor_back.enable(1)
+
+        self.__arm = self.__supervisor.getDevice(f'hand_{self.measuring_side}_Dz')
+
+        self.__node = rclpy.node.Node('webots_resistance_meter_driver')
+        self.__action = ActionServer(
+            self.__node,
+            ResistanceMeter,
+            f'{namespace}/resistance_meter/{self.measuring_side}',
+            execute_callback=self.__execute_callback,
+            callback_group=ReentrantCallbackGroup(),
+            goal_callback=self.__goal_callback,
+            cancel_callback=self.__cancel_callback
+        )
+
+    def __goal_callback(self, _):
+        return GoalResponse.ACCEPT
+
+    def __cancel_callback(self, _):
+        return CancelResponse.ACCEPT
+
+    def __execute_callback(self, goal_handle):
+        result = ResistanceMeter.Result()
+        resistance = 0
+
+        if goal_handle.is_cancel_requested:
+            goal_handle.cancelled()
+
+        position = self.discretize_robot_position()
+        force = self.measure_touch_force()
+
+        if position is not None:
+            if force >= FORCE_TRESHOLD:
+                resistance = EXCAVATION_SQUARES['resistances'][position]
+                noise = randrange(
+                    resistance * MEASUREMENT_NOISE * -1,
+                    resistance * MEASUREMENT_NOISE
+                ) / 100.0
+                inaccuracy = randrange(
+                    resistance * MEASUREMENT_INACCURACY * -1,
+                    resistance * MEASUREMENT_INACCURACY
+                ) / 100.0
+                resistance = int(resistance + noise + inaccuracy)
+
+        result.resistance = resistance
+        goal_handle.succeed()
+
+        return result
+
+    def measure_touch_force(self):
+
+        fork_front = self.__touch_sensor_front.getValues()
+        fork_back = self.__touch_sensor_back.getValues()
+
+        return vector_to_average_scalar(fork_front) + \
+            vector_to_average_scalar(fork_back)
+
+    def discretize_robot_position(self):
+
+        trn = self.__robot.getField('translation').getSFVec3f()
+        rot = self.__robot.getField('rotation').getSFRotation()
+
+        if value_in_range(
+            trn[1],
+            EXCAVATION_SQUARES['y_start'], EXCAVATION_SQUARES['y_end']
+        ):
+
+            # Robot's yaw in degrees
+            y = axangle2euler((rot[0], rot[1], rot[2]), rot[3])[0] * 180 / pi
```

Comment:
```suggestion
            yaw = math.degrees(rot[3])
```
- Uvijek imamo rotaciju oko z ose samo, tako da ti je yaw isto sto i rot[3]
- Za rad <-> deg citljivije je `math.radians/math.degrees`
- y -> yaw, ovako msm da je y koordinata :)